lib/bundler/source/path.rb

```diff
@@ -65,7 +65,7 @@ def hash
 
       def eql?(other)
         return unless other.class == self.class
-        expand(@original_path) == expand(other.original_path) &&
+        expand(@original_path) == other.expanded_original_path &&
```

Comment:
You know... that change would have made sense, huh... oops

I will update when I add a test for this as well.  Thanks.